lib/storage/sfCacheSessionStorage.class.php

```diff
@@ -226,42 +250,44 @@ public function remove($key)
     return $retval;
   }
 
-  /**
-   * Regenerates id that represents this storage.
-   *
-   * @param boolean $destroy Destroy session when regenerating?
-   *
-   * @return boolean True if session regenerated, false if error
-   *
-   * @throws <b>sfStorageException</b> If an error occurs while regenerating this storage
-   */
-  public function regenerate($destroy = false)
-  {
-    if($destroy)
+    /**
+     * Regenerates id that represents this storage.
+     *
+     * @param boolean $destroy Destroy session when regenerating?
+     *
+     * @return boolean True if session regenerated, false if error
+     *
+     * @throws <b>sfStorageException</b> If an error occurs while regenerating this storage
+     */
+    public function regenerate($destroy = false)
```

Comment:
This cannot be added.

`sfCacheSessionStorage::regenerate()` overloads the parent `sfStorage::regenerate()`, which means the method signature must match that of the parent method. Adding a `bool` return type to the overloaded method would break the signature matching and cause an error.